frontend/src/metabase/dashboard/components/Dashboard/Dashboard.unit.spec.tsx

```diff
@@ -0,0 +1,188 @@
+import { renderWithProviders, waitForLoaderToBeRemoved } from "__support__/ui";
+import {
+  createMockDashboardCard,
+  createMockDashboard,
+} from "metabase-types/api/mocks";
+import type { Dashboard, Parameter } from "metabase-types/api";
+import { Dashboard as DashboardComponent } from "./Dashboard";
+
+type SetupOpts = {
+  dashboardId: number | null;
+  dashboard: Dashboard | null;
+  selectedTabId: number | null; // when there's no tabs, is null
+  parameters: Parameter[]; // when empty, is an empty array
+  skipLoader?: boolean;
+  isEditing: boolean;
+};
+
+function dashboardWithCards(first: { isDirty?: boolean } = {}) {
+  return createMockDashboard({
+    dashcards: [
+      createMockDashboardCard({ id: 1, isDirty: first.isDirty } as any),
+      createMockDashboardCard({ id: 2 }),
+    ],
+  });
+}
+
+async function setup(overrides: Partial<SetupOpts> = {}) {
+  const mockDashboard = createMockDashboard({ id: 10 }); // value is irrelevant
```

Comment:
Should we just call `createMockDashboard()` without any args if we don't care about the values?
We can use `mockDashboard.id` below if we need it